repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy

```diff
@@ -50,97 +52,109 @@ class HttpConnectionPoolServiceImplTest {
     // Retrieve the defaults defined in the XSD.
     PoolType poolType = new PoolType()
     Integer POOLU_SO_TIMEOUT = poolType.getHttpSocketTimeout()
+    ConfigurationService configurationService = mock(ConfigurationService)
+    HealthCheckService healthCheckService = mock(HealthCheckService)
 
     @Before
     void setUp() {
+        when(healthCheckService.register()).thenReturn(mock(HealthCheckServiceProxy))
+
         List<PoolType> pools = PoolTypeHelper.createListOfPools(2, 2)
-        PoolType poolType = pools.get(0);
+        PoolType poolType = pools.get(0)
         poolType.setId(POOL1_ID)
         poolType.setDefault(POOL1_DEFAULT)
         poolType.setHttpConnManagerMaxTotal(POOL1_MAX_CON)
         poolType.setHttpSocketTimeout(POOL1_SO_TIMEOUT)
-        poolType = pools.get(1);
+        poolType = pools.get(1)
         poolType.setId(POOL2_ID)
         poolType.setDefault(POOL2_DEFAULT)
         poolType.setHttpConnManagerMaxTotal(POOL2_MAX_CON)
         poolType.setHttpSocketTimeout(POOL2_SO_TIMEOUT)
 
-        poolCfg = new HttpConnectionPoolConfig();
-        poolCfg.pool.addAll(pools);
+        poolCfg = new HttpConnectionPoolConfig()
+        poolCfg.pool.addAll(pools)
 
-        srv = new HttpConnectionPoolServiceImpl(mock(ConfigurationService.class), mock(HealthCheckService.class));
-        srv.configure(poolCfg);
+        srv = new HttpConnectionPoolServiceImpl(configurationService, healthCheckService)
+        srv.configure(poolCfg)
+        srv.with {
+            initialized = true
+        }
     }
 
-    @Test
-    void testGetClient() {
-        HttpClient client = srv.getClient(POOL1_ID).getHttpClient();
-        assertEquals("Should retrieve requested client", POOL1_SO_TIMEOUT, client.getParams().getParameter(CoreConnectionPNames.SO_TIMEOUT));
+    @After
+    void destroy() {
+        srv = new HttpConnectionPoolServiceImpl(configurationService, healthCheckService)
+        srv.destroy()
```

Comment:
I'm not sure why you would create a new instance to call destroy on it after every unit test.